### PR TITLE
update for cedar#794

### PR DESCRIPTION
**Pull request by cdisselkoen**
*Issue #, if available:*

cedar-policy/cedar#794

*Description of changes:*

Also a couple unrelated nits.

CedarJavaFFI tests are currently failing due to #126.

### CODE REVIEW

File: CedarJavaFFI/src/interface_result.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "success")]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
+#[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
+/// As of this writing, this is only used by `json_parse_entity_uid()`.
+/// We could remove this and just specialize the output of `json_parse_entity_uid()`.
```
Comment (khieta) [inline CedarJavaFFI/src/interface_result.rs:24]: Do we think we'll add more operations with this return type? If not, I'd lean towards specializing the output of `json_parse_entity_uid()`

**Review by khieta — APPROVED** [review]

File: CedarJavaFFI/src/interface_result.rs (hunk 1 above)
Comment (cdisselkoen) [reply CedarJavaFFI/src/interface_result.rs:24]: I think we'll add more operations in general, in the future.  I am unclear whether they will/should use this return type or not

Comment (khieta) [reply CedarJavaFFI/src/interface_result.rs:24]: A fair question. Let's just leave it in the Java code for now. But for consistency with our other data types, I'm thinking that we just want it to be named `Result` or `Answer` (no "interface"). I'd also prefer not to have the internal/non-internal failure distinction, but I don't feel strongly.

Comment (cdisselkoen) [reply CedarJavaFFI/src/interface_result.rs:24]: Hmm my comment about only used by `json_parse_entity_uid()` appears to be wrong.  I suspect though that post-793 and friends, in some error cases our code is returning the wrong error structure.  Java code probably needs more tests with invalid call JSON.

Comment (cdisselkoen) [reply CedarJavaFFI/src/interface_result.rs]: I think it makes the most sense to just rename `InterfaceResult` to `Answer`, and leave the rest for followups.  I've done that in the latest revision of this PR

**Review by khieta — APPROVED** [review]

**Comment by cdisselkoen** [comment]
I'm not sure what is causing the current CI failure.

File: CedarJavaFFI/src/answer.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "success")]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
```
Comment (john-h-kastner-aws) [inline CedarJavaFFI/src/answer.rs:21]: Should the Java FFI create have these `wasm` attributes?

**Comment by john-h-kastner-aws** [comment]
> I'm not sure what is causing the current CI failure.

It looks like `Build on Linux` is trying to run `/home/runner/.cache/cargo-zigbuild/0.18.3/zigcc-aarch64-apple-darwin.sh`. Could be a problem.

**Comment by cdisselkoen** [comment]
That's super weird because you wouldn't think any of the changes in this PR could be affecting the zig build (I think?)

**Comment by john-h-kastner-aws** [comment]
Zig did release a new version 4 days ago: https://ziglang.org/download/0.12.0/release-notes.html

**Comment by john-h-kastner-aws** [comment]
Confirmed build works on Linux with zig 0.11.0 from https://ziglang.org/download/

**Comment by cdisselkoen** [comment]
Alright, build has passed now, after I both downgraded to zig 0.11.0 and also locked the version of cargo-zigbuild to the current latest for good measure.

**Review by john-h-kastner-aws — APPROVED** [review]
It works!

**Pull request merged**